src/Api/Controllers/AuthRequestsController.cs

```diff
@@ -0,0 +1,155 @@
+﻿using Bit.Api.Models.Request;
+using Bit.Api.Models.Response;
+using Bit.Core.Context;
+using Bit.Core.Entities;
+using Bit.Core.Exceptions;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using Bit.Core.Settings;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Bit.Api.Controllers;
+
+[Route("auth-requests")]
+[Authorize("Application")]
+public class AuthRequestsController : Controller
+{
+    private readonly IUserRepository _userRepository;
+    private readonly IDeviceRepository _deviceRepository;
+    private readonly IUserService _userService;
+    private readonly IAuthRequestRepository _authRequestRepository;
+    private readonly ICurrentContext _currentContext;
+    private readonly IPushNotificationService _pushNotificationService;
+    private readonly IGlobalSettings _globalSettings;
+
+    public AuthRequestsController(
+        IUserRepository userRepository,
+        IDeviceRepository deviceRepository,
+        IUserService userService,
+        IAuthRequestRepository authRequestRepository,
+        ICurrentContext currentContext,
+        IPushNotificationService pushNotificationService,
+        IGlobalSettings globalSettings)
+    {
+        _userRepository = userRepository;
+        _deviceRepository = deviceRepository;
+        _userService = userService;
+        _authRequestRepository = authRequestRepository;
+        _currentContext = currentContext;
+        _pushNotificationService = pushNotificationService;
+        _globalSettings = globalSettings;
+    }
+
+    [HttpGet("")]
+    public async Task<ListResponseModel<AuthRequestResponseModel>> Get()
+    {
+        var userId = _userService.GetProperUserId(User).Value;
+        var authRequests = await _authRequestRepository.GetManyByUserIdAsync(userId);
+        var responses = authRequests.Select(a => new AuthRequestResponseModel(a, _globalSettings.SelfHosted)).ToList();
+        return new ListResponseModel<AuthRequestResponseModel>(responses);
+    }
+
+    [HttpGet("{id}")]
+    public async Task<AuthRequestResponseModel> Get(string id)
+    {
+        var userId = _userService.GetProperUserId(User).Value;
+        var authRequest = await _authRequestRepository.GetByIdAsync(new Guid(id));
+        if (authRequest == null || authRequest.UserId != userId)
+        {
+            throw new NotFoundException();
+        }
+
+        return new AuthRequestResponseModel(authRequest, _globalSettings.SelfHosted);
+    }
+
+    [HttpGet("{id}/response")]
+    [AllowAnonymous]
+    public async Task<AuthRequestResponseModel> GetResponse(string id, [FromQuery] string code)
+    {
+        var authRequest = await _authRequestRepository.GetByIdAsync(new Guid(id));
+        if (authRequest == null || code != authRequest.AccessCode)
+        {
+            throw new NotFoundException();
+        }
+
+        if (authRequest.GetExpirationDate() < DateTime.UtcNow)
+        {
+            throw new NotFoundException();
+        }
```

Comment:
These throw the same exception, any issue with combining these into one check?

Reply:
Implemented on [[review] Consolidate error conditions](https://github.com/bitwarden/server/pull/2276/commits/2853331550a040068c3af68ae6067ee559f63281)